File: src/main/java/com/telegram/codex/conversation/application/gateway/ReplyGenerationGateway.java
package com.telegram.codex.conversation.application.gateway;

import com.telegram.codex.conversation.application.reply.ReplyResult;

import java.nio.file.Path;
import java.util.List;

public interface ReplyGenerationGateway {

    ReplyResult generateReply(
        String userMessage,
        String conversationState,
        List<Path> imageFilePaths,
        String replyToText,
        String longTermMemory
    );
}

File: src/main/kotlin/com/telegram/codex/conversation/application/gateway/ReplyGenerationGateway.kt
package com.telegram.codex.conversation.application.gateway

import com.telegram.codex.conversation.application.reply.ReplyResult
import java.nio.file.Path

interface ReplyGenerationGateway {
    fun generateReply(
        userMessage: String?,
        conversationState: String?,
        imageFilePaths: List<Path>,
        replyToText: String?,
        longTermMemory: String?,
    ): ReplyResult
}
